src/main.rs

```diff
@@ -1,22 +1,34 @@
 mod cmd;
-use std::env;
+use clap::{Parser, Subcommand};
 
-fn main() {
-    let args: Vec<String> = env::args().collect();
-    run(args)
+#[derive(Debug, Parser)]
+#[clap(name = "gh default-branch", version)]
+#[clap(about = "GitHub CLI extension to show & rename the default branch", long_about = None)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+#[derive(Debug, Subcommand)]
+enum Commands {
+    /// Show default branch
+    Show {
+        /// Show only the branch name (e.g. main)
+        #[clap(short, long)]
+        name_only: bool,
```

Comment:
:memo: If I add `[clap(arg_required_else_help = true)]` as follows, everything will be required and help will be displayed if not given.

```rs
    /// Show default branch
    #[clap(arg_required_else_help = true)]
    Show {
        /// Show only the branch name (e.g. main)
        #[clap(short, long)]
        name_only: bool,
    },
```

Note that this is not the intended behavior.

```
$ gh-default-branch show
gh-default-branch-show 
Show default branch

USAGE:
    gh-default-branch show [OPTIONS]

OPTIONS:
    -h, --help         Print help information
    -n, --name-only    Show only the branch name (e.g. main)
```